modules/userdirectory/src/main/java/org/opencastproject/userdirectory/JpaUserReferenceProvider.java

```diff
@@ -159,11 +177,14 @@ public String toString() {
    */
   @Override
   public List<Role> getRolesForUser(String userName) {
-    ArrayList<Role> roles = new ArrayList<Role>();
-    User user = loadUser(userName);
-    if (user != null)
-      roles.addAll(user.getRoles());
-    return roles;
+    if (roleProvider == null) {
```

Comment:
Easier to swap this (less nesting):
```suggestion
    if (roleProvider != null) {
      return roleProvider.getRolesForUser(userName);
    }
```